### ISSUE TITLE
Error message while creating container: The requested resource is in use

### ISSUE THREAD

**Issue by ajkauffmann** [microsoft/navcontainerhelper#1099]
**Describe the issue**
I keep getting an error 'The requested resource is in use'.
Script worked fine last week, but whatever I try now, it fails.

Tried with and without -imageName parameter. Same error message, but in different places. Output for both statements is included.

**Scripts used to create container and cause the issue**
```
New-BcContainer `
-accept_eula `
-accept_outdated `
-containerName $container.ContainerName `
-imageName $container.ImageName `
-artifactUrl $artifactUrl `
-license $container.LicenseFile `
-assignPremiumPlan `
-Credential $ContainerCredential `
-auth NavUserPassword `
-updateHosts `
-dns '8.8.8.8' `
-isolation hyperv `
-shortcuts None `
-doNotCheckHealth `
-myscripts @("C:\Temp\AdditionalOutput.ps1", "c:\Temp\SetupCertificate.ps1", "C:\Temp\docker.local.pfx") `
-useSSL `
-additionalParameters $AdditionalParameters `
-alwaysPull
```

**Full output of scripts**
Output with -imageName:
```
NavContainerHelper is version 0.7.0.13
NavContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 2004
Docker Client Version is 19.03.5
Docker Server Version is 19.03.5
Removing C:\ProgramData\NavContainerHelper\Extensions\bconprem
ArtifactUrl and ImageName specified
Building image bconpremx:onprem-16.3.14085.14238-w1 based on https://bcartifacts.azureedge.net
/onprem/16.3.14085.14238/w1
Pulling latest image mcr.microsoft.com/dynamicsnav:10.0.19041.329-generic
10.0.19041.329-generic: Pulling from dynamicsnav
Generic Tag: 0.1.0.7
Container OS Version: 10.0.19041.329 (2004)
Host OS Version: 10.0.19041.329 (2004)
Copying Platform Artifacts
Copy Database
Copy Licensefile
c:\$TMP$-bconpremx-onprem-16.3.14085.14238-w1
Sending build context to Docker daemon  1.314GB

Step 1/6 : FROM mcr.microsoft.com/dynamicsnav:10.0.19041.329-generic
 ---> d2fd79450ea3
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSa
ndbox=N artifactUrl=https://bcartifacts.azureedge.net/onprem/16.3.14085.14238/w1
 ---> Using cache
 ---> 26b68e700256
Step 3/6 : COPY my /run/
 ---> Using cache
 ---> f9fa4436f028
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> 50b88a50c5a0
Step 5/6 : RUN \Run\start.ps1 -installOnly
 ---> Running in e639048d659b
docker : hcsshim::CreateComputeSystem e639048d659b7823de1d2151a2be57f69db81e3fb0965ba6e0327e4
c6028a51b: The requested resource is in use.
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.13\ContainerHandling\N
ew-NavImage.ps1:294 char:1
+ docker build --isolation=$isolation --memory $memory --tag $imageName ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (hcsshim::Create...urce is in use.:String) [], R 
   emoteException
    + FullyQualifiedErrorId : NativeCommandError
```

Output without -imageName:
```
NavContainerHelper is version 0.7.0.13
NavContainerHelper is running as administrator
Host is Microsoft Windows 10 Pro - 2004
Docker Client Version is 19.03.5
Docker Server Version is 19.03.5
Removing C:\ProgramData\NavContainerHelper\Extensions\bconprem
Pulling image mcr.microsoft.com/dynamicsnav:10.0.19041.329-generic
10.0.19041.329-generic: Pulling from dynamicsnav
Using image mcr.microsoft.com/dynamicsnav:10.0.19041.329-generic
Disabling Health Check (always report healthy)
Creating Container bconprem
Version: 16.3.14085.14238-w1
Style: onprem
Generic Tag: 0.1.0.7
Container OS Version: 10.0.19041.329 (2004)
Host OS Version: 10.0.19041.329 (2004)
Using locale en-US
Using hyperv isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to en
able)
Using license file C:\Users\ajkau\OneDrive\MyDev\Licenses\Kauffmann - 6581468 - BC16 W1.flf
Additional Parameters:
--hostname bconprem.docker.local
Files in C:\ProgramData\NavContainerHelper\Extensions\bconprem\my:
- AdditionalOutput.ps1
- docker.local.pfx
- license.flf
- MainLoop.ps1
- SetupCertificate.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bconprem from image mcr.microsoft.com/dynamicsnav:10.0.19041.329-generic
e6448519d1c503a025a0e087d9baeab5c780e3874301e0c4c052bf4e7e7917b5
DockerDo : docker.exe: Error response from daemon: hcsshim::CreateComputeSystem e6448519d1c50
3a025a0e087d9baeab5c780e3874301e0c4c052bf4e7e7917b5: The requested resource is in use.
(extra info: {"SystemType":"Container","Name":"e6448519d1c503a025a0e087d9baeab5c780e3874301e0
c4c052bf4e7e7917b5","Owner":"docker","IgnoreFlushesDuringBoot":true,"LayerFolderPath":"C:\\Pr
ogramData\\docker\\windowsfilter\\e6448519d1c503a025a0e087d9baeab5c780e3874301e0c4c052bf4e7e7
917b5","Layers":[{"ID":"5f60134d-d60a-5720-8a52-c347c72fdda5","Path":"C:\\ProgramData\\docker
\\windowsfilter\\81251785dde1f4da0a62d0edd7186a72806966e165c87bb93c83e833733eab90"},{"ID":"83
b3d5bb-d56c-5cab-9fb9-f36ea05915e0","Path":"C:\\ProgramData\\docker\\windowsfilter\\7e796ff33
34f02621a9229ef0b18494def918e127f24ee512a1193fe67ab63c6"},{"ID":"e2002363-cac1-5cd8-a375-155b
951930a5","Path":"C:\\ProgramData\\docker\\windowsfilter\\daeb00c0c17e05083c5d28636fd1556aee8
a7f3aaa07269cc31c6171829a6f30"},{"ID":"97e5fe49-9c38-5e15-9e30-c75e3df86030","Path":"C:\\Prog
ramData\\docker\\windowsfilter\\01c8e70cfbf720a2e248f8d3041294471aa3d6e98b5a1eb1c3639b2db7dad
a2d"},{"ID":"ab940034-a781-5e68-b358-85a437fb77f9","Path":"C:\\ProgramData\\docker\\windowsfi
lter\\5b928b6edcd4807d6221de3f31c33713a2c9f113ad4a620baefe9ffae163a749"},{"ID":"f10d84a2-cd24
-5484-a5fc-717f1af609f9","Path":"C:\\ProgramData\\docker\\windowsfilter\\37d17b61f839691a2f7d
0a3102dba8df1bff463be02e7edd5808c9a052451fe2"},{"ID":"e410772a-13b6-5466-8e27-efab1aef5797","
Path":"C:\\ProgramData\\docker\\windowsfilter\\f7c3bc37f5d0ab3a9825a5eabd2d99506c106ae13430e7
fa4941d9b0b70f7d64"},{"ID":"fc4efe0f-dd62-5085-a683-e6f25ede4dd7","Path":"C:\\ProgramData\\do
cker\\windowsfilter\\10aaa123a350fd92c475bbcbf25686b6bc0be3b1851548a1a6060f93e6d14c78"},{"ID"
:"987e8907-0dce-5ad0-bcbb-f7fe8813af50","Path":"C:\\ProgramData\\docker\\windowsfilter\\08e80
c994fa2ccde1b760f76ecc84ce66b4b16c0411e7a622650b0c06c6e4d90"},{"ID":"cf1e66ab-8714-5d41-a7ea-
202e98823310","Path":"C:\\ProgramData\\docker\\windowsfilter\\0374733d56df10c04263f7492e69753
07555a704bd4ac0dd30c1b138b0bee735"},{"ID":"13b2efee-ac17-52e6-a8fc-232fac15b105","Path":"C:\\
ProgramData\\docker\\windowsfilter\\c3edff8ea50413b9d7ac551e376ddc0dca742735a930cab5c18dde652
82d5f8c"},{"ID":"4f6c9847-be9e-573a-a620-c8c2ca1cd4b3","Path":"C:\\ProgramData\\docker\\windo
wsfilter\\384f4df98837363729f74eb9a88e8cc7822471fef364510fa1cac87e7ee47c70"},{"ID":"7743776e-
b428-5381-a6b9-cbc859ea79ae","Path":"C:\\ProgramData\\docker\\windowsfilter\\5bd4ba3718610864
0cb7cc8e78c6fe136cb2a286a6098498e978f5060edb889d"},{"ID":"d4e8bd4b-4655-586d-8e1c-a931fa1dc27
d","Path":"C:\\ProgramData\\docker\\windowsfilter\\0271b32c47bbbc8505acbd275161779de7a62e0499
b931e60da0964a4d73a163"},{"ID":"8f2d63fb-1654-52da-bd73-112df255d74f","Path":"C:\\ProgramData
\\docker\\windowsfilter\\1375ddcf9827035bab20cd74536957a347d74b7b1bf761392969e7ce45a0c954"},{
"ID":"1b356d81-1a4d-5355-91f3-85c8fb2c8bca","Path":"C:\\ProgramData\\docker\\windowsfilter\\c
7fe83f8bf116b0f4543819caca84f9859d1a2a5fd842ddb21157f89e34b420e"},{"ID":"c0d7bf7f-da3e-5763-9
6ee-d93347dde950","Path":"C:\\ProgramData\\docker\\windowsfilter\\7178e7ee1b3265611c8bfe78e3d
756e90b9bfdd444b4e46e640cdff11fc64435"}],"MemoryMaximumInMB":4096,"HostName":"bconprem.docker
.local","MappedDirectories":[{"HostPath":"c:\\windows\\system32\\drivers\\etc","ContainerPath
":"c:\\driversetc","ReadOnly":false,"BandwidthMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":
false},{"HostPath":"c:\\bcartifacts.cache","ContainerPath":"c:\\dl","ReadOnly":false,"Bandwid
thMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false},{"HostPath":"c:\\programdata\\navcont
ainerhelper\\extensions\\bconprem\\my","ContainerPath":"c:\\run\\my","ReadOnly":false,"Bandwi
dthMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false},{"HostPath":"c:\\programdata\\navcon
tainerhelper","ContainerPath":"c:\\programdata\\navcontainerhelper","ReadOnly":false,"Bandwid
thMaximum":0,"IOPSMaximum":0,"CreateInUtilityVM":false}],"HvPartition":true,"EndpointList":["
bb97aaff-da53-458d-b2c7-3e34db096161"],"HvRuntime":{"ImagePath":"C:\\ProgramData\\docker\\win
dowsfilter\\c7fe83f8bf116b0f4543819caca84f9859d1a2a5fd842ddb21157f89e34b420e\\UtilityVM"},"Al
lowUnqualifiedDNSQuery":true}).
ExitCode: 125
Commandline: docker run --dns 8.8.8.8 --no-healthcheck --volume c:\bcartifacts.cache:c:\dl --
label nav=2004 --label cu=cu3 --label version=16.3.14085.14238 --label platform=16.0.14073.14
195 --label country=w1 --env artifactUrl=https://bcartifacts.azureedge.net/onprem/16.3.14085.
14238/w1 --env licenseFile="c:\run\my\license.flf" --name bconprem --hostname bconprem --env 
auth=NavUserPassword --env username="admin" --env ExitOnError=N --env locale=en-US --env data
baseServer="" --env databaseInstance="" --volume "C:\ProgramData\NavContainerHelper:C:\Progra
mData\NavContainerHelper" --volume "C:\ProgramData\NavContainerHelper\Extensions\bconprem\my:
C:\Run\my" --isolation hyperv --restart unless-stopped --memory 4G --env enableApiServices=Y 
--env useSSL=Y --volume "c:\windows\system32\drivers\etc:C:\driversetc" --env securePassword=
76492d1116743f0423413b16050a5345MgB8ADQANwBLAHoAZQBkAGwATgBRAGkASwAzADIAQgAyAFYAeQBaAEsAagByA
EEAPQA9AHwAOABmADgAMQA3AGMAMAAzAGUANgBhAGIAMQA1ADEAYQA4AGIAOQA3ADIAYwA5ADAAMwBlADIAYQBlAGQAMw
AzAGQAMwAyAGYANAAwADEAZAA5AGUAMwAyADkANQA3ADUANQA3AGIAYQBlAGYAMgAyAGEAZQA0AGQAOABiADUANAA= --
env passwordKeyFile="c:\run\my\aes.key" --env removePasswordKeyFile=Y --hostname bconprem.doc
ker.local --env accept_eula=Y --env accept_outdated=Y --detach mcr.microsoft.com/dynamicsnav:
10.0.19041.329-generic
At C:\Program Files\WindowsPowerShell\Modules\navcontainerhelper\0.7.0.13\ContainerHandling\N
ew-NavContainer.ps1:1457 char:19
+ ...       if (!(DockerDo -accept_eula -accept_outdated:$accept_outdated - ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,DockerDo
```

**Additional context**
- does it happen all the time: yes
- did it use to work: yes

**Comment by freddydk** [comment]
I have had that one as well.
Have you tried to reset docker to factory defaults?

**Comment by ajkauffmann** [comment]
I'm not running Docker Desktop, so I don't have a reset to factory feature. Could be done manually of course, but I'm not sure which folders I need to remove.

Found this issue: https://forums.docker.com/t/error-the-requested-resource-is-in-use-only-when-using-memory/95851

Seems to be related to option --memory. And I figured I could try with process isolation to avoid usage of this option. That works fine, so I'm now using process isolation.

**Comment by freddydk** [comment]
https://raw.githubusercontent.com/microsoft/nav-arm-templates/master/CleanupAfterDocker.ps1 does a factory reset of docker enterprise (tested it last week - it still works)

**Comment by freddydk** [comment]
Thanks for the link to docker forum - might come in handy.

**Comment by freddydk** [comment]
Just ran into this one myself on an Azure VM.
Increased the memory on the computer, restarted and it works again.
It is probably trying to allocate a block of memory and is denied that.
Maybe increasing the swap file would also be an option.

**Comment by MiljanSinghammer** [comment]
I just run into the same issue with Docker Desktop after update of to the 20H2 and Docker Desktop 3.0. Factory Reset does not seem to resolve the problem.

**Comment by freddydk** [comment]
How much memory do you have on the machine?

**Comment by miljance** [comment]
I have 64GB and I think that is not the problem. Problem was me and my actions i supose. Reconstruction:
I updated to 20H2 and then I updated the Docker Desktop. The containers weren't working so I figured that I have to turn on the hyperv isolation. At that point I couldn't create new images with that error. After I have entered -isolation process it worked. I am in the process of recreating not only BC but also Base Images and it looks good for the moment. Therefore my non-educated guess is that this error might have something to do with having missmatching isolation along the way.

Recommendation is to stick with the hyperv in order not to get the problem with the next update, right?

**Comment by freddydk** [comment]
My recommendation is to not try and keep containers, but have a script with which you can create a new version of the container.

**Comment by miljance** [comment]
Killing the containers is fine, I wasn't expecting that the images might have to be recreated if I want to switch to hyperv. It also logically ok but I wasn't expecting, it slipped my mind.

**Comment by lvanvugt** [comment]
> I updated to 20H2 and then I updated the Docker Desktop. ...
> After I have entered -isolation process it worked.

For me too, although a factory reset seemed to work at first, but eventually not.

**Comment by staedter** [comment]
We are experiencing this also on one of our developers laptop since a couple of weeks... I am currently trying to troubleshoot the issue. I already factory reset, completely uninstalled and reinstalled docker, Container and Hyper-V Feature, changed data-root, verified that enough RAM was available but nothing works...

Any help would be appreciated

**Comment by miljance** [comment]
> We are experiencing this also on one of our developers laptop since a couple of weeks... I am currently trying to troubleshoot the issue. I already factory reset, completely uninstalled and reinstalled docker, Container and Hyper-V Feature, changed data-root, verified that enough RAM was available but nothing works...
> 
> Any help would be appreciated

Did you try to drop all images and containers and to recreate them using using -isolation process?

**Comment by freddydk** [comment]
try to remove special docker settings (data-root) - and do a docker zap (https://raw.githubusercontent.com/microsoft/nav-arm-templates/master/CleanupAfterDocker.ps1) and check

**Comment by Walwode** [comment]
Hey Freddy, sorry for reopening the topic. My docker was working more or less without major issues since the beginning of AL. Since two weeks (or little bit more) i'm constantly facing issues with creation of docker containers. After rebooting m system, previous created images cannot start anymore. So I have to recreate new containers all the time. Recreating containers ends up in the error "The requested resource is in use". I recognized this always happens when he is downloading a new BC application and platform version. So restarting my windows again, allows me to create a new container. As i'm using insider containers frequently, it's more or less a new container every day.

### Complete reinstall
I have tried to remove docker totally (incl. ci zap) and reinstalled a completely new version, starting to download all from scratch.

### Isolation Process
Isolation process isn't working, because of: 
Container OS Version: 10.0.19042.985 (20H2)
Host OS Version: 10.0.19043.985 (21H1)
WARNING: Host OS and Base Image Container OS doesn't match, defaulting to hyperv. If you do not have Hyper-V installed or you encounter issues, you could try to specify -isolation process

### Unblock
I tried also to unblock all files from `C:\bcartifacts.cache`. Still the same issue. As rebooting (after downloading new BC platform) is solving the issue, it might still be related to it?!

### Specs
My machine has 32GB RAM and about 20GB is always free (according to task manager). Disk space on C: is 70GB+ and on D: 160GB (data-root is `D:\docker`).

Any other ideas about possible solutions?

**Comment by freddydk** [comment]
I had the same issue a few weeks ago, which I also tweeted about.
After upgrading my Host machine to 21H1, I cannot run process isolation anymore and it needs hyperv.
In the later Windows versions, hyperv will allocate space for hyperv in the paging file and I cannot figure out why this doesn't work as expected.
I ended up changing the paging file:
![image](https://user-images.githubusercontent.com/10775043/119605364-f4b16800-bdf0-11eb-8815-1b0bc34a0eb2.png)
It didn't work immediately - but now it works with hyperv and multiple containers.

**Comment by freddydk** [comment]
If this doesn't work - please create a new issue - don't add to old closed images, even though the error message is the same, the root cause might be different.

**Comment by EniShima** [comment]
Hi @freddydk  i had the same issue .  and i can't create any container anymore :(

**Comment by bluehenve** [comment]
I'm sorry to write on a closed posted.
I found it via https://freddysblog.com/2020/10/12/troubleshooting-business-central-on-docker/, and I couldnt find something more relevant :(

I'm a newbie to Docker. But at once I got exited, as Business Central run as a dream at my first attempt with Docker.

But now I keep getting 'The requested resource is in use'.

I have:
Windows 10 Pro
21H1
64-bit 
16,0 GB RAM
Virtual memory: start=10.000MB, MAX=24.000MB
PowerShell 7 version 7.1.3

Docker:
Version 20.10.6
Docker desktop version 3.3.3 (64133)
I've done factory reset via docker desktop
Uninstalled all nessesary programs
Taskmanger says 8 GB available memory

I run this in PowerShell 7:
docker run --name hvBC -e accept_eula=Y -m 4G -e artifacturl=https://bcartifacts.azureedge.net/onprem/17.0.16993.0/us mcr.microsoft.com/businesscentral:10.0.19042.928

Can you help?
Regards, Henrik

**Comment by freddydk** [comment]
Try to use 19041 instead of 19042

Hent Outlook til iOS<https://aka.ms/o0ukef>
________________________________
Fra: bluehenve ***@***.***>
Sendt: Saturday, May 29, 2021 6:09:13 PM
Til: microsoft/navcontainerhelper ***@***.***>
Cc: Freddy Kristiansen ***@***.***>; Mention ***@***.***>
Emne: Re: [microsoft/navcontainerhelper] Error message while creating container: The requested resource is in use (#1099)


I'm sorry to write on a closed posted.
I found it via https://freddysblog.com/2020/10/12/troubleshooting-business-central-on-docker/<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Ffreddysblog.com%2F2020%2F10%2F12%2Ftroubleshooting-business-central-on-docker%2F&data=04%7C01%7Cfreddy.kristiansen%40microsoft.com%7Cdaa513d4920a4078658808d922bc1a8c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637579013587370323%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=XgincPW8KkMJl65twEiig%2BQwPcrFFkTRNTq8goISOKI%3D&reserved=0>, and I couldnt find something more relevant :(

I'm a newbie to Docker. But at once I got exited, as Business Central run as a dream at my first attempt with Docker.

But now I keep getting 'The requested resource is in use'.

I have:
Windows 10 Pro
21H1
64-bit
16,0 GB RAM
Virtual memory: start=10.000MB, MAX=24.000MB
PowerShell 7 version 7.1.3

Docker:
Version 20.10.6
Docker desktop version 3.3.3 (64133)
I've done factory reset via docker desktop
Uninstalled all nessesary programs
Taskmanger says 8 GB available memory

I run this in PowerShell 7:
docker run --name hvBC -e accept_eula=Y -m 4G -e artifacturl=https://bcartifacts.azureedge.net/onprem/17.0.16993.0/us<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fbcartifacts.azureedge.net%2Fonprem%2F17.0.16993.0%2Fus&data=04%7C01%7Cfreddy.kristiansen%40microsoft.com%7Cdaa513d4920a4078658808d922bc1a8c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637579013587380280%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=zImsKRjKN6PyySvSG70BNPMu%2FvhNmn70TutENla7Cmo%3D&reserved=0> mcr.microsoft.com/businesscentral:10.0.19042.928

Can you help?
Regards, Henrik

**Comment by bluehenve** [comment]
Thanks, but alas:

Status: Downloaded newer image for mcr.microsoft.com/businesscentral:10.0.19041.985
docker: Error response from daemon: hcsshim::CreateComputeSystem 1ca4f9c538c6e75b3f76c37759088b864687f2ea653736848e6c20fd78349c96: The requested resource is in use.

Other options?
regards, Henrik

**Comment by bluehenve** [comment]
PS C:\Users\HenrikVestergaard> docker ps -a
CONTAINER ID   IMAGE                                              COMMAND                  CREATED         STATUS    PORTS                                                          NAMES
1ca4f9c538c6   mcr.microsoft.com/businesscentral:10.0.19041.985   "powershell -Command…"   6 minutes ago   Created   80/tcp, 443/tcp, 1433/tcp, 7045-7049/tcp, 7083/tcp, 8080/tcp   hvBC

**Comment by EniShima** [comment]
Hi @freddydk  again . I can't create again the container it show the mesage above: 
![image](https://user-images.githubusercontent.com/25667122/120085492-e86f2880-c0d8-11eb-8d59-2de69ffb3e0e.png)

Please could you help me , i'm blocked :(

**Comment by freddydk** [comment]
Please stop adding to closed issues. Create a new issue.
@bluehenve - add full script + output
@EniShima - https://freddysblog.com/2019/08/04/powershell-for-non-experts/ - the way to add a parameter (-usegenericimage) is not with a , - it is with a ` at the end of the prior line - also add full script + output using copy/paste - not screenshot.

**Comment by EniShima** [comment]
Hi @freddydk  the problem still occurs and i don't know what happened. Sorry that i have commented here but in our company, this is a big issue and we haven't found yet a persistent solution. I just stopped the container and started again the the it show this message :
![image](https://user-images.githubusercontent.com/25667122/124355045-8477dc80-dc0f-11eb-941a-bf9f3e87ebad.png)

**Comment by freddydk** [comment]
You are probably still using hyperv isolation - if you add -isolation process, then it should work.

**Comment by hennord999** [comment]
For me "The requested resource is in use" issue was solved when I realized that I did not have sufficient disk space on my pc and cleaned it up. Apparently it was the requested disk space it could get.

**Comment by lvanvugt** [comment]
Today I found a following work around for "The requested resource is in use" issue when using `-isolation hyperv` (on Win10):

1. Set `-isolation process`
2. Run the script (which will fail with error <need to get the right message>)
3. Revert to `-isolation hyperv` 
4. Run script again

Works for my right now.
